example-2/evm/src/contracts/lendingHub/HubGetters.sol

```diff
@@ -109,27 +119,48 @@ contract HubGetters is Context, HubStructs, HubState {
         for(uint i=0; i<allowList.length; i++) {
             address asset = allowList[i];
 
+            VaultAmount memory normalizedAmounts = getVaultAmounts(vaultOwner, asset);
+
             uint64 price = getOraclePrices(asset);
             
             AssetInfo memory assetInfo = getAssetInfo(asset);
 
-            VaultAmount memory normalizedAmounts = getVaultAmounts(vaultOwner, asset);
-
             AccrualIndices memory indices = getInterestAccrualIndices(asset);
             
             uint256 denormalizedDeposited = denormalizeAmount(normalizedAmounts.deposited, indices.deposited);
             uint256 denormalizedBorrowed = denormalizeAmount(normalizedAmounts.borrowed, indices.borrowed);
 
             effectiveNotionalDeposited += denormalizedDeposited * price / (10**assetInfo.decimals);
-            effectiveNotionalBorrowed += (denormalizedBorrowed + assetAmount) * assetInfo.collateralizationRatio * price / (10**assetInfo.decimals);
-        
-            if(asset == assetAddress){
-                require(assetAmount <= denormalizedDeposited, "Not enough deposited to withdraw");
-                effectiveNotionalDeposited -= assetAmount * price / (10**assetInfo.decimals);
-            }
-        }       
-
-        return (effectiveNotionalDeposited >= effectiveNotionalBorrowed);
+            effectiveNotionalBorrowed += denormalizedBorrowed * assetInfo.collateralizationRatio * price / (10**assetInfo.decimals);
+        }  
+
+        return effectiveNotionalDeposited - effectiveNotionalBorrowed;
+    }
+
+    // TODO: cycle through all assets in the vault
+    function allowedToWithdraw(address vaultOwner, address assetAddress, uint256 assetAmount) internal view returns (bool) {       
+
+        AssetInfo memory assetInfo = getAssetInfo(assetAddress);
+
+        uint64 price = getOraclePrices(assetAddress);
+
+        uint256 vaultValue = getVaultEffectiveNotionalValue(vaultOwner); 
+
+        VaultAmount memory amounts = denormalizeVaultAmount(getVaultAmounts(vaultOwner, assetAddress), assetAddress);
+
+        return (amounts.deposited - amounts.borrowed >= assetAmount) && (vaultValue >= assetAmount * price / (10**assetInfo.decimals));
```

Comment:
should think about: when you try to borrow an asset you have deposited, should it auto withdraw? analogous to: if you try to deposit an asset you have already borrowed, should it auto repay?